server/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -41,13 +41,13 @@ object CORS {
    * Currently, you cannot make permissions depend on request details
    */
   def apply(service: HttpService, config: CORSConfig = DefaultCORSConfig): HttpService = Service.lift { req =>
-    
+
     // In the case of an options request we want to return a simple response with the correct Headers set.
     def createOptionsResponse(origin: Header, acrm: Header): Response = corsHeaders(origin.value, acrm.value)(Response())
 
     def corsHeaders(origin: String, acrm: String)(resp: Response): Response =
       config.allowedHeaders.map(_.mkString("", ", ", "")).fold(resp) { hs =>
-        resp.putHeaders(Header("Access-Control-Allow-Headers", hs))
+        resp.putHeaders(Header("Access-Control-Allow-Headers", hs), Header("Access-Control-Expose-Headers", hs))
```

Comment:
@rossabaker Looks like the browser looks for these headers based on which call is made. If it is get or post, it doesn't even make an options call and excepts these header in the actual call. But in case of PUT, both the pre-flight call and the actual call need these headers. Explained here - https://www.html5rocks.com/en/tutorials/cors/#toc-adding-cors-support-to-the-server